src/xsl/openrosa2html5form.xsl

```diff
@@ -28,20 +28,62 @@ XSLT Stylesheet that transforms OpenRosa style (X)Forms into valid HTMl5 forms
     <xsl:param name="openclinica"/>
     <xsl:output method="html" omit-xml-declaration="yes" encoding="UTF-8" indent="yes"/><!-- for xml: version="1.0" -->
 
+    <xsl:key
+        name="bindings-by-id"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@id" />
+    <xsl:key
+        name="nodeset-bindings"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@nodeset" />
+    <xsl:key
+        name="preload-bindings"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@jr:preload != ''" />
+    <xsl:key
+        name="calculate-bindings"
+        match="/h:html/h:head/xf:model/xf:bind"
+        use="@calculate != ''" />
+    <xsl:key name="instances" match="/h:html/h:head/xf:model/xf:instance" use="true()" />
+    <xsl:key
+        name="primary-instance-root"
```

Comment:
I'm surprised this one makes a difference. Is it requested over and over again? I haven't looked at the context carefully.

Reply:
@eyelidlessness: targeted expressions that are deep and repeatedly used.